### PR TITLE
Feature/howard duplicate hit filter tool updated -- reissuing

### PR DESCRIPTION
**Pull request by brucehoward-physics**
Copying from previous PR:

The intent of this PR is to turn the usage of CollectHits in LArPandora.cxx into an art tool to allow one to do some customization at this stage of input. For example, with this we would plan to have a tool living in icaruscode that basically copies CollectHits but does some filtering while inputting the hits. The default functionality here is intended to effectively preserve previous behavior.

Appreciate eyes on the intended changes. I spotted a few comments and things we probably want to remove before merging, but in any case making the PR now to get people looking at it...

A question: in putting together the art tool, I took some inspiration and guidance from other code. Do we want to actually specify a destructor here? I noticed IShowerTool does, for example, but wasn't sure.

### CODE REVIEW

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionTool.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * 
+ */
+#ifndef LAR_PANDORA_HIT_COLLECTION_TOOL_H
+#define LAR_PANDORA_HIT_COLLECTION_TOOL_H
+
+#include "larpandora/LArPandoraInterface/LArPandoraHelper.h"
+
+#include "art/Framework/Principal/Event.h"
+
+namespace HitCollectionTools {
```
Comment (AndyChappell) [inline larpandora/LArPandoraInterface/LArPandoraHitCollectionTool.h]: I think the namespace here should probably still be lar_pandora.

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
```
Comment (AndyChappell) [inline larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc:1]: I think the _tool suffix can be dropped from the name here and the source file extension should be cxx for consistency with the other larpandora classes. I think this should also be separated into a source and header file to split declarations and definitions.

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *  @brief Implement default hit collection tool
+ * 
+ */
+
+#include "art/Utilities/ToolMacros.h"
+
+#include "larpandora/LArPandoraInterface/LArPandoraHelper.h"
+#include "larpandora/LArPandoraInterface/LArPandoraHitCollectionTool.h"
+
+namespace HitCollectionTools {
```
Comment (AndyChappell) [inline larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc]: As previously, namespace lar_pandora

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  public:
+    explicit LArPandoraHitCollectionToolDefault(const fhicl::ParameterSet& pset);
+    void CollectHits(const art::Event& evt, const std::string& label, lar_pandora::HitVector& hitVector) override;
+  };
+
+  LArPandoraHitCollectionToolDefault::LArPandoraHitCollectionToolDefault(const fhicl::ParameterSet& pset){}
+
+  void LArPandoraHitCollectionToolDefault::CollectHits(const art::Event& evt, const std::string& label, lar_pandora::HitVector& hitVector)
+  {
+    lar_pandora::LArPandoraHelper::CollectHits(evt, label, hitVector);
```
Comment (AndyChappell) [inline larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc]: Won't need the lar_pandora namespace specifcation if the namespace is changed.

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionTool.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  class HitCollectionTool
```
Comment (AndyChappell) [inline larpandora/LArPandoraInterface/LArPandoraHitCollectionTool.h]: Probably want to call this IHitCollectionTool in keeping with other pure interfaces in larpandora.

**Review by AndyChappell — CHANGES_REQUESTED** [review]
Thanks Bruce. I think the logic here looks sound, I'm just trying to get this checked through the high stats CI at the moment, but in the interim I've suggested a few fairly superficial changes.

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc (hunk 2 above)
Comment (brucehoward-physics) [reply larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc:1]: Are you sure I can drop this and it's not required by the art tool?

**Comment by brucehoward-physics** [comment]
Because I was working in a different version than this rebased one, I made edits there and cherry-picked the commit over to this branch. Hopefully that works fine?

Anyway, I tried to address most of the stuff, please let me know if the result is not satisfactory.

For one of the comments I have a follow-up question (about the "_tool.cc" vs. ".cxx")

File: larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc (hunk 2 above)
Comment (AndyChappell) [reply larpandora/LArPandoraInterface/LArPandoraHitCollectionToolDefault_tool.cc:1]: Ah ok, if it's an art requirement then you can ignore my comment.

**Review by AndyChappell — APPROVED** [review]

**Pull request merged**